Some tests failing when running locally in Python 3.13 environment

Raising as part of openjournals/joss-reviews#7947

# Expected Behavior

All tests pass.

# Actual Behavior

Some (64) tests failed.  

<details>

<summary>pytest short test summary info output</summary>

```
=================================================================================================================== short test summary info ====================================================================================================================
FAILED tests/test_kernel_approximation.py::TestSkLearn::test_compatible_estimator[RandomFourierKernelApprox(random_state=1234)-check_n_features_in_after_fitting] - AssertionError: `RandomFourierKernelApprox.transform()` does not check for consistency between input number
FAILED tests/test_kernel_approximation.py::TestSkLearn::test_compatible_estimator[RandomFourierKernelApprox(method='weight_only',random_state=1234)-check_n_features_in_after_fitting] - AssertionError: `RandomFourierKernelApprox.transform()` does not check for consistency between input number
FAILED tests/test_kernel_approximation.py::TestSkLearn::test_compatible_estimator[RandomBinningKernelApprox(random_state=1234)-check_n_features_in_after_fitting] - AssertionError: `RandomBinningKernelApprox.transform()` does not check for consistency between input number
FAILED tests/test_koopman_pipeline.py::TestSkLearn::test_compatible_estimator[KoopmanPipeline(regressor=Edmd())-check_do_not_raise_errors_in_init_or_set_params] - ValueError: not enough values to unpack (expected 2, got 1)
FAILED tests/test_koopman_pipeline.py::TestSkLearn::test_compatible_estimator[KoopmanPipeline(regressor=Edmd())-check_n_features_in_after_fitting] - AssertionError: `KoopmanPipeline.predict()` does not check for consistency between input number
FAILED tests/test_koopman_pipeline.py::TestSkLearn::test_compatible_estimator[KoopmanPipeline(lifting_functions=[('pl',PolynomialLiftingFn())],regressor=Edmd())-check_do_not_raise_errors_in_init_or_set_params] - ValueError: not enough values to unpack (expected 2, got 1)
FAILED tests/test_koopman_pipeline.py::TestSkLearn::test_compatible_estimator[KoopmanPipeline(lifting_functions=[('pl',PolynomialLiftingFn())],regressor=Edmd())-check_n_features_in_after_fitting] - AssertionError: `KoopmanPipeline.predict()` does not check for consistency between input number
FAILED tests/test_koopman_pipeline.py::TestSkLearn::test_compatible_estimator[SplitPipeline()-check_do_not_raise_errors_in_init_or_set_params] - ValueError: not enough values to unpack (expected 2, got 1)
FAILED tests/test_koopman_pipeline.py::TestSkLearn::test_compatible_estimator[SplitPipeline()-check_n_features_in_after_fitting] - AssertionError: `SplitPipeline.transform()` does not check for consistency between input number
FAILED tests/test_koopman_pipeline.py::TestSkLearn::test_compatible_estimator[SplitPipeline(lifting_functions_state=[('pl',PolynomialLiftingFn())])-check_do_not_raise_errors_in_init_or_set_params] - ValueError: not enough values to unpack (expected 2, got 1)
FAILED tests/test_koopman_pipeline.py::TestSkLearn::test_compatible_estimator[SplitPipeline(lifting_functions_state=[('pl',PolynomialLiftingFn())])-check_n_features_in_after_fitting] - AssertionError: `SplitPipeline.transform()` does not check for consistency between input number
FAILED tests/test_lifting_functions.py::TestSkLearn::test_compatible_estimator[PolynomialLiftingFn()-check_n_features_in_after_fitting] - AssertionError: `PolynomialLiftingFn.transform()` does not check for consistency between input number
FAILED tests/test_lifting_functions.py::TestSkLearn::test_compatible_estimator[DelayLiftingFn()-check_n_features_in_after_fitting] - AssertionError: `DelayLiftingFn.transform()` does not check for consistency between input number
FAILED tests/test_lifting_functions.py::TestSkLearn::test_compatible_estimator[SkLearnLiftingFn(transformer=MaxAbsScaler())-check_n_features_in_after_fitting] - AssertionError: `SkLearnLiftingFn.transform()` does not check for consistency between input number
FAILED tests/test_lifting_functions.py::TestSkLearn::test_compatible_estimator[BilinearInputLiftingFn()-check_n_features_in_after_fitting] - AssertionError: `BilinearInputLiftingFn.transform()` does not check for consistency between input number
FAILED tests/test_lifting_functions.py::TestSkLearn::test_compatible_estimator[RbfLiftingFn(centers=QmcCenters(random_state=1234))-check_n_features_in_after_fitting] - AssertionError: `RbfLiftingFn.transform()` does not check for consistency between input number
FAILED tests/test_lifting_functions.py::TestSkLearn::test_compatible_estimator[ConstantLiftingFn()-check_n_features_in_after_fitting] - AssertionError: `ConstantLiftingFn.transform()` does not check for consistency between input number
FAILED tests/test_lifting_functions.py::TestSkLearn::test_compatible_estimator[KernelApproxLiftingFn(kernel_approx=RandomFourierKernelApprox(random_state=1234))-check_n_features_in_after_fitting] - AssertionError: `KernelApproxLiftingFn.transform()` does not check for consistency between input number
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[Edmd()-check_estimator_tags_renamed] - TypeError: Estimator Edmd has defined either `_more_tags` or `_get_tags`, but not `__sklearn_tags__`. If you're customizing tags, and need to support multiple scikit-learn versions, you can implement both `__sklearn_tags__` and `_more_tags` or `_get_t...
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[Edmd()-check_n_features_in_after_fitting] - AssertionError: `Edmd.predict()` does not check for consistency between input number
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[EdmdMeta()-check_estimator_tags_renamed] - TypeError: Estimator EdmdMeta has defined either `_more_tags` or `_get_tags`, but not `__sklearn_tags__`. If you're customizing tags, and need to support multiple scikit-learn versions, you can implement both `__sklearn_tags__` and `_more_tags` or `_g...
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[EdmdMeta()-check_n_features_in_after_fitting] - AssertionError: `EdmdMeta.predict()` does not check for consistency between input number
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[EdmdMeta(regressor=Ridge(alpha=1))-check_estimator_tags_renamed] - TypeError: Estimator EdmdMeta has defined either `_more_tags` or `_get_tags`, but not `__sklearn_tags__`. If you're customizing tags, and need to support multiple scikit-learn versions, you can implement both `__sklearn_tags__` and `_more_tags` or `_g...
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[EdmdMeta(regressor=Ridge(alpha=1))-check_fit_score_takes_y] - ValueError: Found input variables with inconsistent numbers of samples: [30, 1]
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[EdmdMeta(regressor=Ridge(alpha=1))-check_n_features_in_after_fitting] - AssertionError: `EdmdMeta.predict()` does not check for consistency between input number
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[EdmdMeta(regressor=Ridge(alpha=1))-check_pipeline_consistency] - ValueError: Found input variables with inconsistent numbers of samples: [30, 1]
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[EdmdMeta(regressor=Ridge(alpha=1))-check_regressors_train] - AssertionError
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[EdmdMeta(regressor=Ridge(alpha=1))-check_regressors_train(readonly_memmap=True)] - AssertionError
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[EdmdMeta(regressor=Ridge(alpha=1))-check_regressors_train(readonly_memmap=True,X_dtype=float32)] - AssertionError
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[EdmdMeta(regressor=Ridge(alpha=1))-check_methods_sample_order_invariance] - IndexError: index 5 is out of bounds for axis 0 with size 1
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[Dmdc()-check_estimator_tags_renamed] - TypeError: Estimator Dmdc has defined either `_more_tags` or `_get_tags`, but not `__sklearn_tags__`. If you're customizing tags, and need to support multiple scikit-learn versions, you can implement both `__sklearn_tags__` and `_more_tags` or `_get_t...
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[Dmdc()-check_n_features_in_after_fitting] - AssertionError: `Dmdc.predict()` does not check for consistency between input number
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[Dmd()-check_estimator_tags_renamed] - TypeError: Estimator Dmd has defined either `_more_tags` or `_get_tags`, but not `__sklearn_tags__`. If you're customizing tags, and need to support multiple scikit-learn versions, you can implement both `__sklearn_tags__` and `_more_tags` or `_get_ta...
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[Dmd()-check_fit_score_takes_y] - ValueError: matmul: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (n?,k),(k,m?)->(n?,m?) (size 1 is different from 3)
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[Dmd()-check_estimators_overwrite_params] - ValueError: matmul: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (n?,k),(k,m?)->(n?,m?) (size 1 is different from 2)
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[Dmd()-check_dont_overwrite_parameters] - ValueError: matmul: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (n?,k),(k,m?)->(n?,m?) (size 1 is different from 3)
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[Dmd()-check_estimators_fit_returns_self] - ValueError: matmul: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (n?,k),(k,m?)->(n?,m?) (size 1 is different from 2)
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[Dmd()-check_readonly_memmap_input] - ValueError: matmul: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (n?,k),(k,m?)->(n?,m?) (size 1 is different from 2)
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[Dmd()-check_n_features_in_after_fitting] - ValueError: matmul: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (n?,k),(k,m?)->(n?,m?) (size 1 is different from 4)
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[Dmd()-check_positive_only_tag_during_fit] - AssertionError: Estimator 'Dmd' raised ValueError unexpectedly. This happens when passing negative input values as X. If negative values are not supported for this estimator instance, then the tags.input_tags.positive_only tag needs to be set to True.
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[Dmd()-check_estimators_dtypes] - ValueError: matmul: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (n?,k),(k,m?)->(n?,m?) (size 1 is different from 5)
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[Dmd()-check_dtype_object] - ValueError: matmul: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (n?,k),(k,m?)->(n?,m?) (size 1 is different from 10)
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[Dmd()-check_pipeline_consistency] - ValueError: matmul: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (n?,k),(k,m?)->(n?,m?) (size 1 is different from 3)
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[Dmd()-check_estimators_nan_inf] - ValueError: matmul: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (n?,k),(k,m?)->(n?,m?) (size 1 is different from 3)
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[Dmd()-check_estimators_pickle] - ValueError: matmul: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (n?,k),(k,m?)->(n?,m?) (size 1 is different from 3)
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[Dmd()-check_estimators_pickle(readonly_memmap=True)] - ValueError: matmul: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (n?,k),(k,m?)->(n?,m?) (size 1 is different from 3)
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[Dmd()-check_f_contiguous_array_estimator] - ValueError: matmul: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (n?,k),(k,m?)->(n?,m?) (size 1 is different from 3)
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[Dmd()-check_regressors_train] - ValueError: matmul: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (n?,k),(k,m?)->(n?,m?) (size 1 is different from 10)
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[Dmd()-check_regressors_train(readonly_memmap=True)] - ValueError: matmul: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (n?,k),(k,m?)->(n?,m?) (size 1 is different from 10)
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[Dmd()-check_regressors_train(readonly_memmap=True,X_dtype=float32)] - ValueError: matmul: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (n?,k),(k,m?)->(n?,m?) (size 1 is different from 10)
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[Dmd()-check_regressor_data_not_an_array] - ValueError: matmul: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (n?,k),(k,m?)->(n?,m?) (size 1 is different from 10)
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[Dmd()-check_regressor_multioutput] - ValueError: matmul: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (n?,k),(k,m?)->(n?,m?) (size 5 is different from 10)
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[Dmd()-check_regressors_no_decision_function] - ValueError: matmul: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (n?,k),(k,m?)->(n?,m?) (size 1 is different from 4)
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[Dmd()-check_regressors_int] - ValueError: matmul: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (n?,k),(k,m?)->(n?,m?) (size 1 is different from 10)
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[Dmd()-check_methods_sample_order_invariance] - ValueError: matmul: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (n?,k),(k,m?)->(n?,m?) (size 1 is different from 3)
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[Dmd()-check_methods_subset_invariance] - ValueError: matmul: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (n?,k),(k,m?)->(n?,m?) (size 1 is different from 3)
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[Dmd()-check_dict_unchanged] - ValueError: matmul: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (n?,k),(k,m?)->(n?,m?) (size 1 is different from 3)
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[Dmd()-check_fit_idempotent] - ValueError: matmul: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (n?,k),(k,m?)->(n?,m?) (size 1 is different from 2)
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[Dmd()-check_fit_check_is_fitted] - ValueError: matmul: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (n?,k),(k,m?)->(n?,m?) (size 1 is different from 2)
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[Dmd()-check_n_features_in] - ValueError: matmul: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (n?,k),(k,m?)->(n?,m?) (size 1 is different from 2)
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[Dmd()-check_fit2d_predict1d] - ValueError: matmul: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (n?,k),(k,m?)->(n?,m?) (size 1 is different from 3)
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[DataRegressor()-check_estimator_tags_renamed] - TypeError: Estimator DataRegressor has defined either `_more_tags` or `_get_tags`, but not `__sklearn_tags__`. If you're customizing tags, and need to support multiple scikit-learn versions, you can implement both `__sklearn_tags__` and `_more_tags` o...
FAILED tests/test_regressors.py::TestSkLearn::test_compatible_estimator[DataRegressor()-check_n_features_in_after_fitting] - AssertionError: `DataRegressor.predict()` does not check for consistency between input number
FAILED tests/test_util.py::TestSkLearn::test_compatible_estimator[AnglePreprocessor()-check_n_features_in_after_fitting] - AssertionError: `AnglePreprocessor.transform()` does not check for consistency between input number
========================================================================================== 64 failed, 1819 passed, 474 deselected, 1889 warnings in 412.30s (0:06:52) ==========================================================================================
```
</details>

# Steps to Reproduce the Problem

From a clone of repository at commit 2215358

1. Create a blank Python 3.13 virtual environment - I used `uv` - and activate
   ```
   uv venv --python=3.13
   source .venv/bin/activate
   ```
2. Install development dependencies and local package
   ```
   uv pip install -r requirements-dev.txt
   uv pip install .
   ```
3. Run tests (skipping MOSEK dependent tests)
   ```
   pytest tests -k "not mosek"
   ```

# Specifications

- Package version: Current `main` ([2215358](https://github.com/decargroup/pykoop/commit/22153581483c1d36afc4226e45a1efa42d464bf2))
- Python version: 3.13
- Platform: Ubuntu 24.04

Looks like this is not specific to Python 3.13, as I also get failures when running in a Python 3.11 environment:

```
uv venv --python=3.11
source .venv/bin/activate
uv pip install -r requirements-dev.txt
uv pip install .
pytest tests -k "not mosek" -x
```

fails and exits with

```
FAILED tests/test_kernel_approximation.py::TestSkLearn::test_compatible_estimator[RandomFourierKernelApprox(random_state=1234)-check_n_features_in_after_fitting] - AssertionError: `RandomFourierKernelApprox.transform()` does not check for consistency between input number
```

There are also a series of deprecation warnings raised:

<details>

<summary>Full pytest error traceback info</summary>

```
____________________________________________________________________ TestSkLearn.test_compatible_estimator[RandomFourierKernelApprox(random_state=1234)-check_n_features_in_after_fitting] _____________________________________________________________________

name = 'RandomFourierKernelApprox', estimator_orig = RandomFourierKernelApprox(random_state=1234)

    @ignore_warnings(category=FutureWarning)
    def check_n_features_in_after_fitting(name, estimator_orig):
        # Make sure that n_features_in are checked after fitting
        tags = get_tags(estimator_orig)
    
        is_supported_X_types = tags.input_tags.two_d_array or tags.input_tags.categorical
    
        if not is_supported_X_types or tags.no_validation:
            return
    
        rng = np.random.RandomState(0)
    
        estimator = clone(estimator_orig)
        set_random_state(estimator)
        if "warm_start" in estimator.get_params():
            estimator.set_params(warm_start=False)
    
        n_samples = 10
        X = rng.normal(size=(n_samples, 4))
        X = _enforce_estimator_tags_X(estimator, X)
    
        if is_regressor(estimator):
            y = rng.normal(size=n_samples)
        else:
            y = rng.randint(low=0, high=2, size=n_samples)
        y = _enforce_estimator_tags_y(estimator, y)
    
        err_msg = (
            "`{name}.fit()` does not set the `n_features_in_` attribute. "
            "You might want to use `sklearn.utils.validation.validate_data` instead "
            "of `check_array` in `{name}.fit()` which takes care of setting the "
            "attribute.".format(name=name)
        )
    
        estimator.fit(X, y)
        assert hasattr(estimator, "n_features_in_"), err_msg
        assert estimator.n_features_in_ == X.shape[1], err_msg
    
        # check methods will check n_features_in_
        check_methods = [
            "predict",
            "transform",
            "decision_function",
            "predict_proba",
            "score",
        ]
        X_bad = X[:, [1]]
    
        err_msg = """\
            `{name}.{method}()` does not check for consistency between input number
            of features with {name}.fit(), via the `n_features_in_` attribute.
            You might want to use `sklearn.utils.validation.validate_data` instead
            of `check_array` in `{name}.fit()` and {name}.{method}()`. This can be done
            like the following:
            from sklearn.utils.validation import validate_data
            ...
            class MyEstimator(BaseEstimator):
                ...
                def fit(self, X, y):
                    X, y = validate_data(self, X, y, ...)
                    ...
                    return self
                ...
                def {method}(self, X):
                    X = validate_data(self, X, ..., reset=False)
                    ...
                return X
        """
        err_msg = textwrap.dedent(err_msg)
    
        msg = f"X has 1 features, but \\w+ is expecting {X.shape[1]} features as input"
        for method in check_methods:
            if not hasattr(estimator, method):
                continue
    
            callable_method = getattr(estimator, method)
            if method == "score":
                callable_method = partial(callable_method, y=y)
    
            with raises(
                ValueError, match=msg, err_msg=err_msg.format(name=name, method=method)
            ):
>               callable_method(X_bad)

.venv/lib/python3.11/site-packages/sklearn/utils/estimator_checks.py:4501: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
.venv/lib/python3.11/site-packages/sklearn/utils/_set_output.py:319: in wrapped
    data_to_wrap = f(self, X, *args, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = RandomFourierKernelApprox(random_state=0)
X = array([[ 0.40015721],
       [-0.97727788],
       [ 0.4105985 ],
       [ 0.12167502],
       [-0.20515826],
       [ 0.6536186 ],
       [-1.45436567],
       [ 1.46935877],
       [-1.98079647],
       [ 1.20237985]])

    def transform(self, X: np.ndarray) -> np.ndarray:
        """Transform data.
    
        Parameters
        ----------
        X : np.ndarray
            Data matrix.
    
        Returns
        -------
        np.ndarray
            Transformed data matrix.
        """
        sklearn.utils.validation.check_is_fitted(self)
        X = sklearn.utils.validation.check_array(X)
        X_scaled = np.sqrt(2 * self.shape) * X
>       products = X_scaled @ self.random_weights_  # (n_samples, n_components)
E       ValueError: matmul: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (n?,k),(k,m?)->(n?,m?) (size 4 is different from 1)

pykoop/kernel_approximation.py:271: ValueError

The above exception was the direct cause of the following exception:

self = <test_kernel_approximation.TestSkLearn object at 0x7b7a3a8e7210>, estimator = RandomFourierKernelApprox(random_state=1234), check = functools.partial(<function check_n_features_in_after_fitting at 0x7b7a3ab4f060>, 'RandomFourierKernelApprox')

    @sklearn.utils.estimator_checks.parametrize_with_checks([
        pykoop.RandomFourierKernelApprox(
            method='weight_offset',
            random_state=1234,
        ),
        pykoop.RandomFourierKernelApprox(
            method='weight_only',
            random_state=1234,
        ),
        pykoop.RandomBinningKernelApprox(random_state=1234),
    ])
    def test_compatible_estimator(self, estimator, check):
        """Test ``scikit-learn`` compatibility of estimators."""
>       check(estimator)

tests/test_kernel_approximation.py:435: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
.venv/lib/python3.11/site-packages/sklearn/utils/_testing.py:147: in wrapper
    return fn(*args, **kwargs)
.venv/lib/python3.11/site-packages/sklearn/utils/estimator_checks.py:4498: in check_n_features_in_after_fitting
    with raises(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <sklearn.utils._testing._Raises object at 0x7b7a204dbb50>, exc_type = <class 'ValueError'>
exc_value = ValueError('matmul: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (n?,k),(k,m?)->(n?,m?) (size 4 is different from 1)'), _ = <traceback object at 0x7b7a202080c0>

    def __exit__(self, exc_type, exc_value, _):
        # see
        # https://docs.python.org/2.5/whatsnew/pep-343.html#SECTION000910000000000000000
    
        if exc_type is None:  # No exception was raised in the block
            if self.may_pass:
                return True  # CM is happy
            else:
                err_msg = self.err_msg or f"Did not raise: {self.expected_exc_types}"
                raise AssertionError(err_msg)
    
        if not any(
            issubclass(exc_type, expected_type)
            for expected_type in self.expected_exc_types
        ):
            if self.err_msg is not None:
                raise AssertionError(self.err_msg) from exc_value
            else:
                return False  # will re-raise the original exception
    
        if self.matches is not None:
            err_msg = self.err_msg or (
                "The error message should contain one of the following "
                "patterns:\n{}\nGot {}".format("\n".join(self.matches), str(exc_value))
            )
            if not any(re.search(match, str(exc_value)) for match in self.matches):
>               raise AssertionError(err_msg) from exc_value
E               AssertionError: `RandomFourierKernelApprox.transform()` does not check for consistency between input number
E               of features with RandomFourierKernelApprox.fit(), via the `n_features_in_` attribute.
E               You might want to use `sklearn.utils.validation.validate_data` instead
E               of `check_array` in `RandomFourierKernelApprox.fit()` and RandomFourierKernelApprox.transform()`. This can be done
E               like the following:
E               from sklearn.utils.validation import validate_data
E               ...
E               class MyEstimator(BaseEstimator):
E                   ...
E                   def fit(self, X, y):
E                       X, y = validate_data(self, X, y, ...)
E                       ...
E                       return self
E                   ...
E                   def transform(self, X):
E                       X = validate_data(self, X, ..., reset=False)
E                       ...
E                   return X

.venv/lib/python3.11/site-packages/sklearn/utils/_testing.py:1114: AssertionError
```

</details>

There are also a series of `sklearn` deprecation warnings raised, which possibly points to this being due to some breaking change in `sklearn`?

--- a/.github/workflows/test-package-no-mosek.yml
+++ b/.github/workflows/test-package-no-mosek.yml
@@ -10,7 +10,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ['3.8', '3.9', '3.10', '3.11']
+        python-version: ['3.8', '3.9', '3.10', '3.11', '3.12']
     steps:
     - uses: actions/checkout@v2
     - name: Set up Python ${{matrix.python-version}}

--- a/.github/workflows/test-package.yml
+++ b/.github/workflows/test-package.yml
@@ -10,7 +10,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ['3.12']
+        python-version: ['3.13']
     steps:
     - uses: actions/checkout@v2
     - name: Set up Python ${{matrix.python-version}}

--- a/pykoop/_sklearn_metaestimators/metaestimators.py
+++ b/pykoop/_sklearn_metaestimators/metaestimators.py
@@ -15,6 +15,7 @@ for the full license.
 """
 
 import abc
+from contextlib import suppress
 from typing import Any, Dict, List
 
 import numpy as np
@@ -28,10 +29,18 @@ class _BaseComposition(sklearn.base.BaseEstimator, metaclass=abc.ABCMeta):
         out = super().get_params(deep=deep)
         if not deep:
             return out
+
         estimators = getattr(self, attr)
-        if not hasattr(estimators, '__iter__'):
+        try:
+            out.update(estimators)
+        except (TypeError, ValueError):
+            # Ignore TypeError for cases where estimators is not a list of
+            # (name, estimator) and ignore ValueError when the list is not
+            # formatted correctly. This is to prevent errors when calling
+            # `set_params`. `BaseEstimator.set_params` calls `get_params` which
+            # can error for invalid values for `estimators`.
             return out
-        out.update(estimators)
+
         for name, estimator in estimators:
             if hasattr(estimator, "get_params"):
                 for key, value in estimator.get_params(deep=True).items():
@@ -43,14 +52,18 @@ class _BaseComposition(sklearn.base.BaseEstimator, metaclass=abc.ABCMeta):
         # 1. All steps
         if attr in params:
             setattr(self, attr, params.pop(attr))
-        # 2. Step replacement
+        # 2. Replace items with estimators in params
         items = getattr(self, attr)
-        names = []
-        if hasattr(items, '__iter__'):
-            names, _ = zip(*items)
-        for name in list(params.keys()):
-            if "__" not in name and name in names:
-                self._replace_estimator(attr, name, params.pop(name))
+        if isinstance(items, list) and items:
+            # Get item names used to identify valid names in params
+            # `zip` raises a TypeError when `items` does not contains
+            # elements of length 2
+            with suppress(TypeError):
+                item_names, _ = zip(*items)
+                for name in list(params.keys()):
+                    if "__" not in name and name in item_names:
+                        self._replace_estimator(attr, name, params.pop(name))
+
         # 3. Step parameters and other initialisation arguments
         super().set_params(**params)
         return self
@@ -68,11 +81,11 @@ class _BaseComposition(sklearn.base.BaseEstimator, metaclass=abc.ABCMeta):
         if len(set(names)) != len(names):
             raise ValueError("Names provided are not unique: {0!r}".format(
                 list(names)))
-        conflict_names = set(names).intersection(self.get_params(deep=False))
-        if conflict_names:
+        invalid_names = set(names).intersection(self.get_params(deep=False))
+        if invalid_names:
             raise ValueError(
                 "Estimator names conflict with constructor arguments: {0!r}".
-                format(sorted(conflict_names)))
+                format(sorted(invalid_names)))
         invalid_names = [name for name in names if "__" in name]
         if invalid_names:
             raise ValueError(

--- a/pykoop/centers.py
+++ b/pykoop/centers.py
@@ -78,7 +78,7 @@ class GridCenters(Centers):
     >>> grid.fit(X_msd[:, 1:])  # Remove episode feature
     GridCenters(n_points_per_feature=4)
     >>> grid.centers_
-    array([...])
+    array(...)
     """
 
     def __init__(
@@ -155,7 +155,7 @@ class UniformRandomCenters(Centers):
     >>> rand.fit(X_msd[:, 1:])  # Remove episode feature
     UniformRandomCenters(n_centers=10)
     >>> rand.centers_
-    array([...])
+    array(...)
     """
 
     def __init__(
@@ -235,7 +235,7 @@ class GaussianRandomCenters(Centers):
     >>> rand.fit(X_msd[:, 1:])  # Remove episode feature
     GaussianRandomCenters(n_centers=10)
     >>> rand.centers_
-    array([...])
+    array(...)
     """
 
     def __init__(
@@ -306,7 +306,7 @@ class QmcCenters(Centers):
     >>> qmc.fit(X_msd[:, 1:])  # Remove episode feature
     QmcCenters(n_centers=10)
     >>> qmc.centers_
-    array([...])
+    array(...)
 
     Generate centers using a Sobol sequence
 
@@ -314,7 +314,7 @@ class QmcCenters(Centers):
     >>> qmc.fit(X_msd[:, 1:])  # Remove episode feature
     QmcCenters(n_centers=8, qmc=<class 'scipy.stats._qmc.Sobol'>)
     >>> qmc.centers_
-    array([...])
+    array(...)
     """
 
     def __init__(
@@ -430,7 +430,7 @@ class ClusterCenters(Centers):
     >>> kmeans.fit(X_msd[:, 1:])  # Remove episode feature
     ClusterCenters(estimator=KMeans(n_clusters=3))
     >>> kmeans.centers_
-    array([...])
+    array(...)
     """
 
     def __init__(
@@ -507,7 +507,7 @@ class GaussianMixtureRandomCenters(Centers):
     >>> gmm.fit(X_msd[:, 1:])  # Remove episode feature
     GaussianMixtureRandomCenters(estimator=GaussianMixture(n_components=3))
     >>> gmm.centers_
-    array([...])
+    array(...)
     """
 
     def __init__(

--- a/pykoop/kernel_approximation.py
+++ b/pykoop/kernel_approximation.py
@@ -267,6 +267,10 @@ class RandomFourierKernelApprox(KernelApproximation):
         """
         sklearn.utils.validation.check_is_fitted(self)
         X = sklearn.utils.validation.check_array(X)
+        if X.shape[1] != self.n_features_in_:
+            raise ValueError(
+                f"X has {X.shape[1]} features, but {self.__class__.__name__} "
+                f"is expecting {self.n_features_in_} features as input.")
         X_scaled = np.sqrt(2 * self.shape) * X
         products = X_scaled @ self.random_weights_  # (n_samples, n_components)
         if self.method == 'weight_only':
@@ -451,6 +455,10 @@ class RandomBinningKernelApprox(KernelApproximation):
         """
         sklearn.utils.validation.check_is_fitted(self)
         X = sklearn.utils.validation.check_array(X)
+        if X.shape[1] != self.n_features_in_:
+            raise ValueError(
+                f"X has {X.shape[1]} features, but {self.__class__.__name__} "
+                f"is expecting {self.n_features_in_} features as input.")
         X_scaled = np.sqrt(2 * self.shape) * X
         X_hashed = self._hash_samples(X_scaled)
         Xt = self.encoder_.transform(X_hashed) / np.sqrt(self.n_components)

--- a/pykoop/kernel_approximation.py
+++ b/pykoop/kernel_approximation.py
@@ -113,7 +113,7 @@ class RandomFourierKernelApprox(KernelApproximation):
     >>> ka.fit(X_msd[:, 1:])  # Remove episode feature
     RandomFourierKernelApprox(n_components=10, random_state=1234)
     >>> ka.transform(X_msd[:, 1:])
-    array([...])
+    array(...)
     """
 
     _ft_lookup = {
@@ -318,7 +318,7 @@ class RandomBinningKernelApprox(KernelApproximation):
     >>> ka.fit(X_msd[:, 1:])  # Remove episode feature
     RandomBinningKernelApprox(n_components=10, random_state=1234)
     >>> ka.transform(X_msd[:, 1:])
-    array([...])
+    array(...)
     """
 
     _ddot_lookup = {

--- a/pykoop/koopman_pipeline.py
+++ b/pykoop/koopman_pipeline.py
@@ -721,7 +721,7 @@ class EpisodeIndependentLiftingFn(KoopmanLiftingFn):
         }
         # Validate data
         X = sklearn.utils.validation.check_array(X, **self._check_array_params)
-        # Set numbre of input features (including episode feature)
+        # Set number of input features (including episode feature)
         self.n_features_in_ = X.shape[1]
         # Extract episode feature
         if self.episode_feature_:
@@ -751,12 +751,11 @@ class EpisodeIndependentLiftingFn(KoopmanLiftingFn):
                 X,
                 **self._check_array_params,
             )
-            # Check input shape
+            # Check number of features
             if X.shape[1] != self.n_features_in_:
-                raise ValueError(f'{self.__class__.__name__} `fit()` called '
-                                 f'with {self.n_features_in_} features, but '
-                                 f'`transform()` called with {X.shape[1]} '
-                                 'features.')
+                raise ValueError(
+                    f"X has {X.shape[1]} features, but {self.__class__.__name__} "
+                    f"is expecting {self.n_features_in_} features as input.")
         return self._apply_transform_or_inverse(X, 'transform')
 
     def inverse_transform(self, X: np.ndarray) -> np.ndarray:
@@ -996,12 +995,11 @@ class EpisodeDependentLiftingFn(KoopmanLiftingFn):
                 X,
                 **self._check_array_params,
             )
-            # Check input shape
+            # Check number of features
             if X.shape[1] != self.n_features_in_:
-                raise ValueError(f'{self.__class__.__name__} `fit()` called '
-                                 f'with {self.n_features_in_} features, but '
-                                 f'`transform()` called with {X.shape[1]} '
-                                 'features.')
+                raise ValueError(
+                    f"X has {X.shape[1]} features, but {self.__class__.__name__} "
+                    f"is expecting {self.n_features_in_} features as input.")
         return self._apply_transform_or_inverse(X, 'transform')
 
     def inverse_transform(self, X: np.ndarray) -> np.ndarray:
@@ -1014,12 +1012,11 @@ class EpisodeDependentLiftingFn(KoopmanLiftingFn):
                 X,
                 **self._check_array_params,
             )
-            # Check input shape
+            # Check number of features
             if X.shape[1] != self.n_features_out_:
-                raise ValueError(f'{self.__class__.__name__} `fit()` output '
-                                 f'{self.n_features_out_} features, but '
-                                 '`inverse_transform()` called with '
-                                 f'{X.shape[1]} features.')
+                raise ValueError(
+                    f"X has {X.shape[1]} features, but {self.__class__.__name__} "
+                    f"is expecting {self.n_features_out_} features as input.")
         return self._apply_transform_or_inverse(X, 'inverse_transform')
 
     def _apply_transform_or_inverse(self, X: np.ndarray,
@@ -1270,6 +1267,11 @@ class KoopmanRegressor(
         self._validate_feature_names(X)
         # Validate array
         X = sklearn.utils.validation.check_array(X, **self._check_array_params)
+        # Check number of features
+        if X.shape[1] != self.n_features_in_:
+            raise ValueError(
+                f"X has {X.shape[1]} features, but {self.__class__.__name__} "
+                f"is expecting {self.n_features_in_} features as input.")
         # Split episodes
         episodes = split_episodes(X, episode_feature=self.episode_feature_)
         # Predict for each episode
@@ -1623,8 +1625,12 @@ class KoopmanRegressor(
         if not np.all(_extract_feature_names(X) == self.feature_names_in_):
             raise ValueError('Input features do not match fit features.')
 
-    # Extra estimator tags
-    # https://scikit-learn.org/stable/developers/develop.html#estimator-tags
+    def __sklearn_tags__(self):
+        tags = super().__sklearn_tags__()
+        tags.target_tags.single_output = False
+        tags.target_tags.multi_output = True
+        return tags
+
     def _more_tags(self):
         return {
             'multioutput': True,
@@ -1812,12 +1818,11 @@ class SplitPipeline(metaestimators._BaseComposition, KoopmanLiftingFn):
                 X,
                 **self._check_array_params,
             )
-            # Check input shape
+            # Check number of features
             if X.shape[1] != self.n_features_in_:
-                raise ValueError(f'{self.__class__.__name__} `fit()` called '
-                                 f'with {self.n_features_in_} features, but '
-                                 f'`transform()` called with {X.shape[1]} '
-                                 'features.')
+                raise ValueError(
+                    f"X has {X.shape[1]} features, but {self.__class__.__name__} "
+                    f"is expecting {self.n_features_in_} features as input.")
         # Split episodes
         episodes = split_episodes(X, episode_feature=self.episode_feature_)
         episodes_state = []
@@ -1873,12 +1878,11 @@ class SplitPipeline(metaestimators._BaseComposition, KoopmanLiftingFn):
             sklearn.utils.validation.check_is_fitted(self)
             X = sklearn.utils.validation.check_array(
                 X, **self._check_array_params)
-            # Check input shape
+            # Check number of features
             if X.shape[1] != self.n_features_out_:
-                raise ValueError(f'{self.__class__.__name__} `fit()` output '
-                                 f'{self.n_features_out_} features, but '
-                                 '`inverse_transform()` called with '
-                                 f'{X.shape[1]} features.')
+                raise ValueError(
+                    f"X has {X.shape[1]} features, but {self.__class__.__name__} "
+                    f"is expecting {self.n_features_out_} features as input.")
         # Split episodes
         episodes = split_episodes(X, episode_feature=self.episode_feature_)
         episodes_state = []
@@ -2278,12 +2282,11 @@ class KoopmanPipeline(metaestimators._BaseComposition, KoopmanLiftingFn):
                 X,
                 **self._check_array_params,
             )
-            # Check input shape
+            # Check number of features
             if X.shape[1] != self.n_features_in_:
-                raise ValueError(f'{self.__class__.__name__} `fit()` called '
-                                 f'with {self.n_features_in_} features, but '
-                                 f'`transform()` called with {X.shape[1]} '
-                                 'features.')
+                raise ValueError(
+                    f"X has {X.shape[1]} features, but {self.__class__.__name__} "
+                    f"is expecting {self.n_features_in_} features as input.")
         # Apply lifting functions
         X_out = X
         for _, lf in self.lifting_functions_:
@@ -2309,12 +2312,11 @@ class KoopmanPipeline(metaestimators._BaseComposition, KoopmanLiftingFn):
                 X,
                 **self._check_array_params,
             )
-            # Check input shape
+            # Check number of features
             if X.shape[1] != self.n_features_out_:
-                raise ValueError(f'{self.__class__.__name__} `fit()` output '
-                                 f'{self.n_features_out_} features, but '
-                                 '`inverse_transform()` called with '
-                                 f'{X.shape[1]} features.')
+                raise ValueError(
+                    f"X has {X.shape[1]} features, but {self.__class__.__name__} "
+                    f"is expecting {self.n_features_out_} features as input.")
         # Apply inverse lifting functions in reverse order
         X_out = X
         for _, lf in self.lifting_functions_[::-1]:
@@ -2360,6 +2362,11 @@ class KoopmanPipeline(metaestimators._BaseComposition, KoopmanLiftingFn):
                 X,
                 **self._check_array_params,
             )
+            # Check number of features
+            if X.shape[1] != self.n_features_in_:
+                raise ValueError(
+                    f"X has {X.shape[1]} features, but {self.__class__.__name__} "
+                    f"is expecting {self.n_features_in_} features as input.")
         # Lift data matrix
         X_trans = self.transform(X)
         # Predict in lifted space
@@ -2405,6 +2412,11 @@ class KoopmanPipeline(metaestimators._BaseComposition, KoopmanLiftingFn):
         self._validate_feature_names(X)
         # Validate input array
         X = sklearn.utils.validation.check_array(X, **self._check_array_params)
+        # Check number of features
+        if X.shape[1] != self.n_features_in_:
+            raise ValueError(
+                f"X has {X.shape[1]} features, but {self.__class__.__name__} "
+                f"is expecting {self.n_features_in_} features as input.")
         scorer = KoopmanPipeline.make_scorer()
         score = scorer(self, X, None)
         return score

--- a/pykoop/koopman_pipeline.py
+++ b/pykoop/koopman_pipeline.py
@@ -1627,6 +1627,7 @@ class KoopmanRegressor(
 
     def __sklearn_tags__(self):
         tags = super().__sklearn_tags__()
+        tags.target_tags.required = False
         tags.target_tags.single_output = False
         tags.target_tags.multi_output = True
         return tags

--- a/pykoop/lifting_functions.py
+++ b/pykoop/lifting_functions.py
@@ -72,9 +72,9 @@ class SkLearnLiftingFn(koopman_pipeline.EpisodeIndependentLiftingFn):
     ['ep', 'StandardScaler(x0)', 'StandardScaler(x1)', 'StandardScaler(u0)']
     >>> X_msd_pp = std_scaler.transform(X_msd)
     >>> np.mean(X_msd_pp[:, 1:], axis=0)
-    array([...])
+    array(...)
     >>> np.std(X_msd_pp[:, 1:], axis=0)
-    array([...])
+    array(...)
     """
 
     def __init__(

--- a/pykoop/lmi_regressors.py
+++ b/pykoop/lmi_regressors.py
@@ -99,6 +99,7 @@ class LmiRegressor(koopman_pipeline.KoopmanRegressor):
 
     def __sklearn_tags__(self):
         tags = super().__sklearn_tags__()
+        tags.target_tags.required = False
         tags.target_tags.single_output = False
         tags.target_tags.multi_output = True
         # Hard to guarantee exact idempotence when calling external solver.
@@ -2382,6 +2383,7 @@ class LmiHinfZpkMeta(sklearn.base.RegressorMixin, sklearn.base.BaseEstimator):
 
     def __sklearn_tags__(self):
         tags = super().__sklearn_tags__()
+        tags.target_tags.required = False
         tags.target_tags.single_output = False
         tags.target_tags.multi_output = True
         return tags

--- a/pykoop/lmi_regressors.py
+++ b/pykoop/lmi_regressors.py
@@ -97,6 +97,14 @@ class LmiRegressor(koopman_pipeline.KoopmanRegressor):
         'dtype': 'float64',
     }
 
+    def __sklearn_tags__(self):
+        tags = super().__sklearn_tags__()
+        tags.target_tags.single_output = False
+        tags.target_tags.multi_output = True
+        # Hard to guarantee exact idempotence when calling external solver.
+        tags.non_deterministic = True
+        return tags
+
     def _more_tags(self):
         reason = ('Hard to guarantee exact idempotence when calling external '
                   'solver.')
@@ -2371,6 +2379,12 @@ class LmiHinfZpkMeta(sklearn.base.RegressorMixin, sklearn.base.BaseEstimator):
         # Ensure fit has been done
         sklearn.utils.validation.check_is_fitted(self)
         return self.hinf_regressor_.predict(X)
+
+    def __sklearn_tags__(self):
+        tags = super().__sklearn_tags__()
+        tags.target_tags.single_output = False
+        tags.target_tags.multi_output = True
+        return tags
 
     def _more_tags(self):
         return {

--- a/pykoop/regressors.py
+++ b/pykoop/regressors.py
@@ -149,7 +149,7 @@ class EdmdMeta(koopman_pipeline.KoopmanRegressor):
         G = (Theta_p @ Psi.T) / q
         H = (Psi @ Psi.T) / q
         self.regressor_.fit(H.T, G.T)
-        coef = self.regressor_.coef_.T
+        coef = np.atleast_2d(self.regressor_.coef_).T
         return coef
 
     def _validate_parameters(self) -> None:
@@ -395,6 +395,16 @@ class Dmd(koopman_pipeline.KoopmanRegressor):
         if self.mode_type not in valid_mode_types:
             raise ValueError(f'`mode_type` must be one of {valid_mode_types}')
 
+    def __sklearn_tags__(self):
+        tags = super().__sklearn_tags__()
+        tags.target_tags.single_output = False
+        tags.target_tags.multi_output = True
+        # The `dmd.Dmd` class requires X and y to have the samenumber of
+        # features. This test does not meet that requirement and must be
+        # skipped for now.
+        tags._skip_test = True
+        return tags
+
     def _more_tags(self):
         reason = ('The `dmd.Dmd` class requires X and y to have the same '
                   'number of features. This test does not meet that '
@@ -500,6 +510,15 @@ class DataRegressor(koopman_pipeline.KoopmanRegressor):
     def _validate_parameters(self) -> None:
         # No parameters to validate
         pass
+
+    def __sklearn_tags__(self):
+        tags = super().__sklearn_tags__()
+        tags.target_tags.single_output = False
+        tags.target_tags.multi_output = True
+        # Allow a bad score since the ``coef_`` matrix will be filled with
+        # zeros, and we just care to test ``scikit-learn`` API compliance.
+        tags.regressor_tags.poor_score = True
+        return tags
 
     def _more_tags(self):
         return {

--- a/pykoop/regressors.py
+++ b/pykoop/regressors.py
@@ -397,6 +397,7 @@ class Dmd(koopman_pipeline.KoopmanRegressor):
 
     def __sklearn_tags__(self):
         tags = super().__sklearn_tags__()
+        tags.target_tags.required = False
         tags.target_tags.single_output = False
         tags.target_tags.multi_output = True
         # The `dmd.Dmd` class requires X and y to have the samenumber of
@@ -513,6 +514,7 @@ class DataRegressor(koopman_pipeline.KoopmanRegressor):
 
     def __sklearn_tags__(self):
         tags = super().__sklearn_tags__()
+        tags.target_tags.required = False
         tags.target_tags.single_output = False
         tags.target_tags.multi_output = True
         # Allow a bad score since the ``coef_`` matrix will be filled with

--- a/pykoop/tsvd.py
+++ b/pykoop/tsvd.py
@@ -79,7 +79,7 @@ class Tsvd(sklearn.base.BaseEstimator):
         >>> tsvd.fit(X_msd)
         Tsvd()
         >>> tsvd.singular_values_
-        array([...])
+        array(...)
 
         SVD with cutoff truncation
 
@@ -87,7 +87,7 @@ class Tsvd(sklearn.base.BaseEstimator):
         >>> tsvd.fit(X_msd)
         Tsvd(truncation='cutoff', truncation_param=0.001)
         >>> tsvd.singular_values_
-        array([...])
+        array(...)
 
         SVD with manual rank truncation
 
@@ -95,7 +95,7 @@ class Tsvd(sklearn.base.BaseEstimator):
         >>> tsvd.fit(X_msd)
         Tsvd(truncation='rank', truncation_param=2)
         >>> tsvd.singular_values_
-        array([...])
+        array(...)
         """
         self.truncation = truncation
         self.truncation_param = truncation_param

--- a/pykoop/util.py
+++ b/pykoop/util.py
@@ -458,8 +458,8 @@ def example_data_pendulum() -> Dict[str, Any]:
     X_pend = np.vstack(X_pend_lst)
     valid_ep = [5, 25, 45]
     train_ep = list(set(range(n_ep)) - set(valid_ep))
-    valid_idx = np.where(np.in1d(X_pend[:, 0], valid_ep))[0]
-    train_idx = np.where(np.in1d(X_pend[:, 0], train_ep))[0]
+    valid_idx = np.where(np.isin(X_pend[:, 0], valid_ep))[0]
+    train_idx = np.where(np.isin(X_pend[:, 0], train_ep))[0]
     X_train = X_pend[train_idx, :]
     X_valid = X_pend[valid_idx, :]
     n_inputs = 0
@@ -536,8 +536,8 @@ def example_data_duffing() -> Dict[str, Any]:
     X_do = np.vstack(X_do_lst)
     valid_ep = [n_ep - 3, n_ep - 2, n_ep - 1]
     train_ep = list(set(range(n_ep)) - set(valid_ep))
-    valid_idx = np.where(np.in1d(X_do[:, 0], valid_ep))[0]
-    train_idx = np.where(np.in1d(X_do[:, 0], train_ep))[0]
+    valid_idx = np.where(np.isin(X_do[:, 0], valid_ep))[0]
+    train_idx = np.where(np.isin(X_do[:, 0], train_ep))[0]
     X_train = X_do[train_idx, :]
     X_valid = X_do[valid_idx, :]
     n_inputs = 1

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,7 +1,7 @@
 # Requirements to install ``pykoop`` and run examples
 numpy>=1.21.0
 scipy>=1.8.0
-scikit-learn>=1.2.0
+scikit-learn>=1.2.0, <1.7.0
 PICOS>=2.4.0
 optht>=0.2.0
 Deprecated>=1.2.13

--- a/setup.py
+++ b/setup.py
@@ -39,7 +39,7 @@ setuptools.setup(
     install_requires=[
         'numpy>=1.21.0',
         'scipy>=1.7.0',
-        'scikit-learn>=1.0.0',
+        'scikit-learn>=1.2.0, <1.7.0',
         'picos>=2.4.0',
         'optht>=0.2.0',
         'Deprecated>=1.2.13',

